ankaios_sdk/_components/request.py

```diff
@@ -113,71 +121,138 @@ def get_id(self) -> str:
+            self, complete_state: CompleteState, masks: list = []
+            ) -> None:
         """
-        Sets the update masks for the request.
+        Initializes an UpdateStateRequest instance.
 
         Args:
+            complete_state (CompleteState): The new state to set.
             masks (list): The masks to set for the request.
         """
-        if self._request_type == RequestType.UPDATE_STATE:
-            self._request.updateStateRequest.updateMask[:] = masks
-        elif self._request_type == RequestType.GET_STATE:
-            self._request.completeStateRequest.fieldMask[:] = masks
+        super().__init__()
+        self._request.updateStateRequest.updateMask[:] = masks
+        self._request.updateStateRequest.newState.CopyFrom(
+            complete_state._to_proto()
+        )
 
-    def _to_proto(self) -> _ank_base.Request:
-        """
-        Converts the Request object to a proto message.
+        self.logger.debug("Created request of type UpdateState with id %s",
+                          self._request.requestId)
 
-        Returns:
-            _ank_base.Request: The protobuf message representing the request.
-        """
-        return self._request
 
+# pylint: disable=too-few-public-methods, dangerous-default-value
+class LogsRequest(Request):
+    """
+    Represents a request for getting logs from the Ankaios system.
+    """
+    def __init__(
+            self, workload_names: list[WorkloadInstanceName], follow: bool = False,
+            tail: int = -1, since: Union[str, datetime] = "",
+            until: Union[str, datetime] = "",
+            ) -> None:
+        """
+        Initializes an LogsRequest instance.
 
-class RequestType(Enum):
-    """ Enumeration for the different types of requests. """
-    UPDATE_STATE = 1
-    "(int): Request for updating the state."
-    GET_STATE = 2
-    "(int): Request for getting the state."
+        Args:
+            workload_names (list[WorkloadInstanceName]): The workload instance
+                names for which to get logs.
+            follow (bool): If true, the logs will be continuously streamed.
+            tail (int): The number of lines to display from the end of the logs.
+            since (str / datetime): The start time for the logs. If string, it must
+                be in the RFC3339 format.
+            until (str / datetime): The end time for the logs. If string, it must
+                be in the RFC3339 format.
 
-    def __str__(self) -> str:
+        Raises:
+            ValueError: If no workload names are provided.
         """
-        Return the string representation of the enum value.
+        if len(workload_names) == 0:
+            raise ValueError("At least one workload name must be provided.")
+
+        super().__init__()
+        self._request.logsRequest = _ank_base.LogsRequest(
+            workloadNames=[name._to_proto() for name in workload_names],
+            follow=follow,
+            tail=tail
+        )
```

Comment:
You cannot set `self._request.logsRequest.`.

```suggestion
        self._request.logsRequest.CopyFrom(_ank_base.LogsRequest(
            workloadNames=[name._to_proto() for name in workload_names],
            follow=follow,
            tail=tail
        ))
```